neo/SmartContract/NativeContract/NativeNep5Neo.cs

```diff
@@ -0,0 +1,108 @@
+
+        public string Description => "";
+
+
+        public bool Contract_Main(ExecutionEngine engine)
+        {
+            //throw new NotImplementedException();
+            var trigger = (engine.Service as StandardService).Trigger;
+            if (trigger == TriggerType.Verification)
+            {
+                StackItem returnvalue = false;
+                engine.CurrentContext.EvaluationStack.Push(returnvalue);
+                return true;
+            }
+            else if (trigger == TriggerType.Application)
+            {
+                var name = engine.CurrentContext.EvaluationStack.Pop().GetString();
+                var _params = engine.CurrentContext.EvaluationStack.Pop() as Neo.VM.Types.Array;
+                if (name == "name")
+                {
+                    return _Name(engine);
+                }
+                if (name == "decimal")
+                {
+                    return _Decimal(engine);
+                }
+                if (name == "balanceof")
+                {
+                    var from = _params[0].GetByteArray();
+                    return _BalanceOf(engine, from);
+                }
+                if (name == "transfer")
+                {
+                    var from = _params[0].GetByteArray();
+                    var to = _params[1].GetByteArray();
+                    var value = _params[2].GetBigInteger();
+                    return _Transfer(engine, from, to, value);
+                }
+            }
+            ///其他的nep5接口，待實現
+            return false;
+        }
+        bool _Name(ExecutionEngine engine)
+        {
+            StackItem returnvalue = System.Text.Encoding.UTF8.GetBytes(this.Name);
+            engine.CurrentContext.EvaluationStack.Push(returnvalue);
+            return true;
+        }
+        bool _Decimal(ExecutionEngine engine)
+        {
+            StackItem returnvalue = 8;
+            engine.CurrentContext.EvaluationStack.Push(returnvalue);
+            return true;
+        }
+        bool _BalanceOf(ExecutionEngine engine, byte[] from)
+        {
+            var blance = NativeContractTool.Storage_Get_Current(engine, from);
+            StackItem returnvalue = blance.Value;
+            engine.CurrentContext.EvaluationStack.Push(returnvalue);
+            return true;
+        }
+        bool _Transfer(ExecutionEngine engine, byte[] from, byte[] to, System.Numerics.BigInteger value)
+        {
+
+            //直接操作存儲區
+            var fromvalue = new System.Numerics.BigInteger(NativeContractTool.Storage_Get_Current(engine, from).Value);
+            if (fromvalue < value)
+                return false;
+            NativeContractTool.Storage_Put_Current(engine, from, (fromvalue - value).ToByteArray());
+
+            var tovalue = new System.Numerics.BigInteger(NativeContractTool.Storage_Get_Current(engine, from).Value);
+            NativeContractTool.Storage_Put_Current(engine, to, (tovalue + value).ToByteArray());
```

Comment:
Better create `Storage_Put_Current` overload that receives `BigInteger` directly... no explicit convertion to bytearray will be needed here (it's safer).